prow/cmd/deck/main.go

```diff
@@ -105,10 +109,27 @@ func main() {
 			path: *tideURL,
 		}
 		ta.start()
-		http.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
+		mux.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
 	}
 
-	logger.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+	// optionally inject http->https redirect handler when behind loadbalancer
+	if *redirectHTTPTo != "" {
+		redirectMux := http.NewServeMux()
+		redirectMux.Handle("/", func(oldMux *http.ServeMux, host string) http.HandlerFunc {
+			return func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("x-forwarded-proto") == "http" {
+					redirectURL, _ := url.Parse(r.URL.String())
```

Comment:
we could handle this error and panic but `.String()` and `Parse()` here are both from `net/url` which shouldn't allow this to fail..

Reply:
We don't need to panic we can just return a server error. Something structured like this wouldn't add much verbosity: https://github.com/kubernetes/test-infra/blob/4f52db4c696846352067c65e27e99b9e3ead49cd/prow/pluginhelp/externalplugins/externalplugins.go#L50-L54

Reply:
SGTM

Reply:
done